node_groups.tf

```diff
@@ -19,6 +19,25 @@ locals {
   }
 }
 
+# This sleep resource is used to provide a timed gap between the cluster creation and the downstream dependencies
```

Comment:
Is there any documented reason we cannot use a `depends_on` object on all compute modules referencing `aws_eks_addon.before_compute`? Is this purely to avoid module dependencies?

If there was an actual dependency in place, there could be additional benefits (e.g. creating the VPC CNI with IRSA and eliminating the need for removing any instance roles after cluster creation). Anecdotally `cluster_addon` creation is incredibly variable (I've seen it finish in seconds, and sometime in minutes).